### DISCUSSION TITLE
Jams duration for symbolic datasets

### DISCUSSION THREAD

**Discussion by rabitt** [mir-dataset-loaders/mirdata#450] [API Design]
We currently require Track objects to have a jams_converter method, which is required to generate jams objects which validate. In order for the jams object to validate, it needs to have a "duration" passed. For some symbolic datasets, this doesn't make as much sense, and the jams format in general is not build with symbolic data in mind. Some ideas for solutions:
1. we could remove the jams_converter requirement from Track objects for certain types of datasets, and allow it to raise a custom "NotImplemented" because it's symbolic.
2. we can keep the Track object as-is, and create a new object tailored for symbolic data, e.g. `SymbolicTrack`.

@PRamoneda and @nkundiushuti - I think solution 1 is the simplest, but maybe there's other custom things we want to make consistent for all symbolic data. Or do you think the current Track object is fine?

As a temporary patch, for symbolic datasets we can put a dummy value for duration, like 1.

Thanks @PRamoneda and @nkundiushuti for bringing this issue up!

**Comment by magdalenafuentes** [comment]
I'm not too familiar with symbolic datasets, `duration` does not make sense at all in that context? Couldn't it be obtained from a midi file or something like that? @nkundiushuti @PRamoneda

If `duration` or the `to_jams` don't make sense, then I'm down for @rabitt's solution 1. If they do and `duration` can be obtained from `midi` or so, I would modify `to_jams()` to to get the duration from the `symbolic` attribute of the `track object` (or whatever we decide at the end).

**Comment by rabitt** [comment]
Just to link them, the outcome of this depends on what we decide with #298 !
